documentation/developers/mapping_instructions.md

```diff
@@ -0,0 +1,748 @@
+            "name": "Curly",
+            "number": 2,
+            "house": "2aff48d8-917a-4c9b-814d-99639a220773"
+        },
+        {
+            "id": "767f38e8-ce3c-41ee-bd4f-cc1c41bbbde9",
+            "name": "Moe",
+            "number": 3,
+            "house": "a4354016-1a5a-47ef-9d14-01217f33ff2e"
+        }
+    ]
+    ```
+
+    Notice that Larry and Curly still live together, and Moe still lives alone.
+
+
+
+
+---
+
+<a name="extractor-mechanism"></a>
+## Extractor Mechanism
+
+As stated previously, Mapping instruction sets are executed in order. If the extractor encounters an error on an instruction, it caches that instuction it and continues down the list. Once finished with the list, it will reattempt the instuctions that had errors. It will continue to retry until either all instuctions have been completed _or_ a round occurs where none of the cached erroneous instuctions were able to be completed. If there are errors at the end of extraction, the errors are reported.
+
+Let's introduce some new artifacts so we can illustrate the process from the extractor point of view.
+
+<details open>
+    <summary><i>Input <b>(Hide/Show)</b></i></summary>
+    
+```json
+{
+    "family_name": "Keaton",
+    "address":"1000 Chortle Way",
+    "adults": [
+        {"name": "Buster"},
+        {"name": "Grace"}
+    ],
+    "children" : [
+        {"name": "Ace"},
+        {"name": "Franklin"}
+    ]
+}
+```
+
+</details>
+<details open>
+    <summary><i>Schemas <b>(Hide/Show)</b></i></summary>
+
+```json
+[
+  {
+    "name": "Person",
+    "type": "record",
+    "fields": [
+      {
+        "name": "id",
+        "type": "string",
+        "jsonldPredicate": "@id"
+      },
+      {
+        "name": "name",
+        "type": "string"
+      },
+      {
+        "name": "last_name",
+        "type": "string"
+      },
+      {
+        "name": "house",
+        "type": "string",
+        "jsonldPredicate": {
+          "_id": "eha.aether.jokes.House",
+          "_type": "@id"
+        }
+      }
+    ],
+    "namespace": "eha.aether.jokes"
+  },
+  {
+    "name": "House",
+    "type": "record",
+    "fields": [
+      {
+        "name": "id",
+        "type": "string",
+        "jsonldPredicate": "@id"
+      },
+      {
+        "name": "address",
+        "type": "string"
+      }
+    ],
+    "namespace": "eha.aether.jokes"
+  }
+]
+```
+
+</details>
+
+##### Steps
+
+Let's start with a suboptimal mappings
+
+<details open>
+    <summary><i>Mapping <b>(Hide/Show)</b></i></summary>
+
+```json
+[
+  [
+    "#!entity-reference#$.House[*].id",
+    "Person.house"
+  ],
+  [
+    "$.family_name",
+    "Person.last_name"
+  ],
+  [
+    "$.adults[*].name",
+    "Person.name"
+  ],
+  [
+    "$.children[*].name",
+    "Person.name"
+  ],
+  [
+    "$.address",
+    "House.address"
+  ],
+  [
+    "#!uuid",
+    "Person.id"
+  ],
+  [
+    "#!uuid",
+    "House.id"
+  ]
+]
+```
+
+</details>
+
+---
+<details>
```

Comment:
Do you want it close by default? Or maybe `<details open>`?

Reply:
I say this because the first time I thought it was already open and I was wondering about the small example... :thinking:

Reply:
Might as well start them all open, I suppose.